File: 169-majority-element/majority-element.cpp
class Solution
{
    public:
        int majorityElement(vector<int> &nums)
        {
            int n = nums.size();
            map<int, int> m;
            for (int i = 0; i < n; i++)
            {
                m[nums[i]]++;
            }
            int f = n / 2;
            int ans;

            for (auto it: m)
            {
                if (it.second > f) {
                    ans=it.first;
                }
            }
            return ans;
        }
};
  

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        hash_map={}
        for i in range(len(nums)):
            hash_map[nums[i]]=hash_map.get(nums[i],0)+1
        n=len(nums)
        for k,v in hash_map.items():
            if v>=n/2:
                return k
                
            
        